### ISSUE TITLE
spreadsheet view support for larger data

### ISSUE THREAD

**Issue by cmdcolin** [GMOD/jbrowse-components#660]
It was noted that a bottleneck of the spreadsheet is just large amounts of data on the screen. Also if a spreadsheet view is open it noticeably slows down e.g. the loading progress bar for a track. Closing the spreadsheet/sv inspector view speeds it back up

Noticed with the bedpe file from skbr3 open

**Comment by cmdcolin** [comment]
Ex. https://github.com/bvaughn/react-window

**Comment by rbuels** [comment]
Implementation strategy:
* parse chunks of the CSV/VCF/whatever inside a web worker (make sure you can transfer a local file handle over the RPC boundary)
* use a react-window or similar thing to avoid having tons of DOM nodes in the table

**Comment by rbuels** [comment]
Another issue with the current design of the spreadsheet view is that all the data is kept in the non-volatile state tree, which makes any saved session containing spreadsheet data really big.

**Comment by rbuels** [comment]
Updated implementation sketch:

* use a react-window or similar thing to avoid having tons of DOM nodes in the table
* make a new kind of data adapter base class and API called BaseTabularDataAdapter that defines a new data adapter API for tabular data
* make new rpc calls for fetching ranges of tabular data using these adapters (kind of depends on #818, but not strictly)
* make adapters for the existing tabular formats
* update the spreadsheet view to use these adapters

**Comment by garrettjstevens** [comment]
Under displayMethods (https://github.com/GMOD/jbrowse-components/issues/852), we can envision a Spreadsheet View as being a display method of a track. For example, a Variant Track would have a VCFTabix adapter, and that adapter could be used to show variants in a Linear Genome View track, or to load the rows of a Spreadsheet.

In the spreadsheet wizard, you would choose which track you wanted to use to open the spreadsheet.

**Comment by elliothershberg** [comment]
The Data Grid components from Material UI could be worth thinking about here. Intended to be performant with a lot of data. https://material-ui.com/components/data-grid/

**Comment by rbuels** [comment]
Crazy idea: what if we could bind a Google Sheet to a spreadsheet view, so that updates propagate in both directions, using the sheets API https://developers.google.com/sheets/api

probably also depends on #604

**Comment by cmdcolin** [comment]
Spreadsheets can now support somewhat larger data through the pagination. Making it more scalable still would be good but might need a refreshed plan

**Comment by rbuels** [comment]
superseded by #1549
